shiny/app.py

```diff
@@ -0,0 +1,112 @@
+    ),
+    ui.div(style = "height:30px"),
+)
+
+def server(input, output, session):
+    @reactive.Calc
+    def parsed_file() -> object:
+        file: list[FileInfo] | None = input.repcsv()
+        if file is None:
+            return pd.DataFrame()
+        elif file[0]["datapath"][-3:] == 'csv':
+            return pd.read_csv(file[0]["datapath"])
+        elif file[0]["datapath"][-3:] == 'mat':
+            raw_m, out_m = loadmat(file[0]["datapath"]), dict()
+            nonvars = ['__header__', '__version__', '__globals__']
+            for key in raw_m.keys() - nonvars:
+                num_columns = raw_m[key].shape[1]
+                if num_columns > 1:
+                    for i in range(1, num_columns + 1):
+                        out_m[f'{key}_{i}'] = raw_m[key][:, i - 1]
+                else:
+                    out_m[key] = raw_m[key][:, 0]
+            return pd.DataFrame(out_m)
+        elif file[0]["datapath"][-3:] == 'dta':
+            return pd.read_stata(file[0]["datapath"])
+        elif file[0]["datapath"][-3:].lower() == 'rds':
+            raw_r = read_r(file[0]["datapath"])
+            return raw_r[None]
+        elif file[0]["datapath"][-3:] == 'xls' or file[0]["datapath"][-4:] == 'xlsx':
+            return pd.read_excel(file[0]["datapath"])
+    
+    @reactive.Calc
+    def eval_ooi() -> object:
+        df_rep = parsed_file()
+        if not input.repname() == "":
+            rep = df_rep.loc[:, input.repname()].values
+        else:
+            rep = df_rep.iloc[:, 0].values
+        rep = rep[~np.isnan(rep)]
+        ooi = ObjectOfInterest(input.est(), input.se(), rep)
+        return ooi
+
+    @output
+    @render.table
+    def summary() -> object:
+        df_rep = parsed_file()
+        if df_rep.empty:
+            return pd.DataFrame()
+
+        row_count = df_rep.shape[0]
+        column_count = df_rep.shape[1]
+        names = df_rep.columns.tolist()
+        column_names = ", ".join(str(name) for name in names)
+        info_df = pd.DataFrame({"Row Count": [row_count],
+                                "Column Count": [column_count],
+                                "Column Names": [column_names],})
+        return info_df
+
+    @output
+    @render.plot(alt="pp plot", width = 670, height = 670)
+    def pp_plot() -> object:
+        ooi = eval_ooi()
+        fig = ooi.pp_plot(fontsize = 13.5, legend_fontsize = 14.5, labelsize = 18)
+        return fig
+    
+    @output
+    @render.table
+    def stats() -> object:
+        ooi = eval_ooi()
+        ooi.get_sk_min()
+        df_stats = pd.DataFrame(columns = ['SK distance', 'Minimum SK distance', 'SK minimizing mean', 'SK minimizing SD'])
+        df_stats.at[0, 'SK distance'] = ooi.sk_dist
+        df_stats.at[0, 'Minimum SK distance'] = ooi.skmin
+        df_stats.at[0, 'SK minimizing mean'] = ooi.skmin_mean
+        df_stats.at[0, 'SK minimizing SD'] = ooi.skmin_sd
```

Comment:
I put in the values here:
https://github.com/JMSLab/BootstrapReport/blob/3fb6719ed5d8a090c161c263c63bdf739ec83152/examples/_generate_example_data.py#L43
and used the [`normal_replicates`](https://github.com/JMSLab/BootstrapReport/blob/3fb6719ed5d8a090c161c263c63bdf739ec83152/examples/normal_replicates.csv) as the example data.

I was expecting to get a low SK since I used the mean and SD that generated the data. Instead the SK is pretty large and the p-p plot is very bowed. 

Am I missing something or is it possible there's a bug somewhere in one of the pathways in the code?

![image](https://github.com/JMSLab/BootstrapReport/assets/7840353/aa305b9c-7e83-47f9-8ca2-b55f9fd3bc40)

Reply:
I do not have a great answer for you, besides that I added extra randomness in the data generating process by generating the variables from a $\mathcal{N}(\mu, n \times \sigma)$ distribution:

https://github.com/JMSLab/BootstrapReport/blob/a9710406b7c8720e564e7e3eea5f3c57ef9a019e/examples/_generate_example_data.py#L44-L45

. . . and performing a bootstrap below:

https://github.com/JMSLab/BootstrapReport/blob/a9710406b7c8720e564e7e3eea5f3c57ef9a019e/examples/_generate_example_data.py#L69-L73

So we're relying on (1), a sample of (100) random variables to well-approximate a normal distribution with large variance and (2), a draw of (100) replicates from those random variables to well-approximate the distribution of those (100) random variables.

If we use the mean and sample-std we get a much nicer looking plot:
![image](https://github.com/JMSLab/BootstrapReport/assets/111905359/5bbd2592-2889-480b-8d2c-39f0e04f122c)